test_partner_client.py

```diff
@@ -0,0 +1,131 @@
+
+from datetime import datetime
+import requests
+from requests import ConnectionError, Timeout
+import retry
+import re
+
+# TODO Figure out logging given how Clarity works.
+
+
+class TestPartnerSampleInformation(object):
+
+    def __init__(self, referral_code, lab_referral, arrival_date, result_date, comment, cov19_result):
+
+        # REVIEW Discussion point. Not all values here are required, but I see no reason not
+        #        to send them, since I do thing we should have them for all cases.
+
+        super().__init__()
+
+        if not isinstance(str, referral_code) and not len(referral_code) == 10:
+            raise AssertionError(
+                "referral_code needs to be 10 digit number string.")
+
+        if not (int(referral_code[:-1]) % 10 == int(referral_code[-1])):
+            raise AssertionError("Check code digit {} (last digit) did not match mod10 requierment" +
+                                 "in referral code: {}".format(referral_code[-1], referral_code))
+
+        self.referral_code = referral_code
+
+        if len(lab_referral) > 50:
+            raise AssertionError(
+                "Cannot have more then 50 characters in lab_referral code.")
+
+        # REVIEW When reviewing verify against docs that
+        #        this regex is actually correct.
+        allowed_chars_regex = re.compile(r"[^A-Za-z0-9\s\.-]")
```

Comment:
The name says that this is allowed chars, but it seems to be matching disallowed chars, right (because of the caret). So perhaps rename it to `disallowed_chars_regex`?

Reply:
It's unclear to me if this is correct according to the docs, since the documentation is strangely worded. (`*mellanslag - *`)

Reply:
Yes. I agree, it is a bit odd.

Reply:
I interpret it as allowing white space and -. But I can ask them to clarify this.